### PR TITLE
feat: add GalleryVideo component

### LINKED ISSUE #256: Gallery videos not autoplaying, show play button
- autoplay videos shouldn't show a play pause button
- certain videos aren't autoplaying. Example – the purple watson atom in the left column part of the way down in the [idl gallery](https://www.ibm.com/design/language/gallery/)

### PR DESCRIPTION
**Pull request by vpicone**
closes #256 

This adds a super simple video component for use in the IDL Gallery. It effectively makes them appear as super high quality gifs.

Since the we're putting _videos_ in the _image_ gallery component, I feel weird bending the ImageGallery or Video theme components to fit this scenario.

### CODE REVIEW

**Comment by abbeyhrt** [comment]
Throwing this out there, would we want to change the name of ImageGallery to MediaGallery?(if we still can?)  I feel like that better encapsulates what it does and ImageGalleryImage -> MediaGalleryImage and maybe GalleryVideo -> MediaGalleryVideo? Or just GalleryImage and GalleryVideo?

**Comment by vpicone** [comment]
@abbeyhrt definitely a possibility, it'd technically be a breaking change.

I think this is okay for now considering how simple this component would be to port over if we want to undergo all that refactoring.

File: src/pages/gallery/index.mdx [hunk 1]
```diff
@@ -14,549 +23,513 @@ visible should contribute to the correct total statement.” — Eliot Noyes
...
 <ImageGalleryImage alt="Watson" title="Watson" col={4}>
 
-<Video
-  playsInline
-  autoPlay
-  loop
-  poster="images/IBM_iX_Portraits.jpg)"
-  src="videos/Watson_Avatar_Ambient-square-homepage-purple.mp4"
-/>
+<GalleryVideo src={watsonAvatarAmbient} />
```
Comment (emyarod) [inline src/pages/gallery/index.mdx]: are we removing `poster` from this video?

File: src/pages/gallery/index.mdx [hunk 2]
```diff
@@ -14,549 +23,513 @@ visible should contribute to the correct total statement.” — Eliot Noyes
...
 <ImageGalleryImage alt="IBM Security" title="IBM Security" col={4}>
 
-<Video
-  playsInline
-  autoPlay
-  loop
-  poster="images/IBM_Security_Cloud.jpg)"
-  src="videos/IBM_Security_Cloud.mp4"
-/>
+<GalleryVideo src={ibmSecurityCloud} />
```
Comment (emyarod) [inline src/pages/gallery/index.mdx]: same question about the `poster`

Comment (alisonjoseph) [reply src/pages/gallery/index.mdx]: I think the poster was removed since these are autoplay? @vpicone

**Comment by alisonjoseph** [comment]
I think for a11y we need a pause button on these videos. I think Mike mentioned using the small pause button we have on the homepage video, but we should double check on that before adding anything.  @dakahn

File: src/pages/gallery/index.mdx (hunk 2 above)
Comment (vpicone) [reply src/pages/gallery/index.mdx]: @alisonjoseph yeah, we want to start the video immediately so there’s no advantage to fetching the poster. Could definitely try it on a throttled connection to see if it makes a dif tho.

**Comment by vpicone** [comment]
@alisonjoseph their essentially gifs now. If the user agent has reduce motion on, they won’t auto play so are covered for a11y.

**Comment by vpicone** [comment]
@abbeyhrt per https://www.w3.org/TR/WCAG20-TECHS/F93.html 

>This failure occurs when an audio or video element _with an audio track_ contains the autoplay attribute and does not contain the muted attribute, and no controls or commands have been provided to pause or stop the media resource.

We good

**Pull request merged**